File: Backend/HoPla_API/Entities/Reservation.cs
using System.ComponentModel.DataAnnotations;

namespace HoPla_API.Entities
{
    public class Reservation
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public User? User { get; set; }

        [Required]
        public Item? Item { get; set; }

        [Required]
        public DateTime StartTime { get; set; }

        [Required]
        public DateTime EndTime { get; set; }
    }

    public class ReservationRequest
    {
        public DateTime StartTime { get; set; }

        public DateTime EndTime { get; set; }
    }
}

File: Frontend/lib/model/reservation.dart
import 'package:json_annotation/json_annotation.dart';

import 'hopla_user.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int id;
  User? user;
  DateTime startTime;
  DateTime endTime;

  Reservation({
    required this.id,
    required this.user,
    required this.startTime,
    required this.endTime,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
